sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/MetadataOnlyTable.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connector.catalog;
+
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.spark.annotation.Evolving;
+import org.apache.spark.sql.connector.expressions.Transform;
+import org.apache.spark.sql.types.StructType;
+
+/**
+ * A concrete {@code Table} implementation that only contains the table metadata without
+ * implementing read/write directly. It represents a general Spark data source table or
+ * a Spark view, and relies on Spark to interpret the table metadata, resolve the table
+ * provider into a data source, or read it as a view.
+ * This affects the table read/write operations but not DDL operations.
```

Comment:
> This affects the table read/write operations but not DDL operations.

It seems a bit unclear. Before the change, DDL operations of DSV2 tables relies on Spark too.

Reply:
maybe we should just remove this line? DDL operations do not call read/write APIs of `Table` anyway.